src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteredMSExperiment.cpp

```diff
@@ -0,0 +1,172 @@
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Lars Nilse $
+// $Authors: Lars Nilse $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/KERNEL/StandardTypes.h>
+#include <OpenMS/CONCEPT/Constants.h>
+#include <OpenMS/KERNEL/FeatureHandle.h>
+#include <OpenMS/KERNEL/ConsensusMap.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <OpenMS/FORMAT/ConsensusXMLFile.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/MultiplexSatelliteCentroided.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteredPeak.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteredMSExperiment.h>
+
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+using namespace std;
+
+namespace OpenMS
+{
+
+  MultiplexFilteredMSExperiment::MultiplexFilteredMSExperiment()
+  {
+  }
+
+  void MultiplexFilteredMSExperiment::addPeak(double mz, double rt, size_t mz_idx, size_t rt_idx)
+  {
+    MultiplexFilteredPeak peak(mz, rt, mz_idx, rt_idx);
+    result_.push_back(peak);
+  }
+
+  void MultiplexFilteredMSExperiment::addPeak(MultiplexFilteredPeak peak)
+  {
+    result_.push_back(peak);
+  }
+
+  MultiplexFilteredPeak MultiplexFilteredMSExperiment::getPeak(size_t i) const
+  {
+    return result_[i];
+  }
+
+  double MultiplexFilteredMSExperiment::getMZ(size_t i) const
+  {
+    return result_[i].getMZ();
+  }
+
+  vector<double> MultiplexFilteredMSExperiment::getMZ() const
+  {
+    vector<double> mz;
+    for (size_t i = 0; i < result_.size(); ++i)
+    {
+      mz.push_back(result_[i].getMZ());
+    }
+    return mz;
+  }
+
+  double MultiplexFilteredMSExperiment::getRT(size_t i) const
+  {
+    return result_[i].getRT();
+  }
+
+  vector<double> MultiplexFilteredMSExperiment::getRT() const
+  {
+    vector<double> rt;
+    for (size_t i = 0; i < result_.size(); ++i)
+    {
+      rt.push_back(result_[i].getRT());
+    }
+    return rt;
+  }
+
+  size_t MultiplexFilteredMSExperiment::size() const
+  {
+    return result_.size();
+  }
+  
+  void MultiplexFilteredMSExperiment::writeDebugOutput(const MSExperiment& exp_picked, String debug_out) const
+  {
+    ConsensusMap map;
+    
+    // loop over peaks
+    for (std::vector<MultiplexFilteredPeak>::const_iterator it_peak = result_.begin(); it_peak < result_.end(); ++it_peak)
+    {
+      ConsensusFeature consensus;
+      
+      consensus.setRT(it_peak->getRT());
+      consensus.setMZ(it_peak->getMZ());
+      consensus.setIntensity(1.0);
+      consensus.setCharge(1);
+      consensus.setQuality(1.0);
+      
+      std::multimap<size_t, MultiplexSatelliteCentroided > satellites = it_peak->getSatellites();
+      int count = 0;
+      
+      // loop over satellites
+      for (std::multimap<size_t, MultiplexSatelliteCentroided >::const_iterator it_satellite = satellites.begin(); it_satellite != satellites.end(); ++it_satellite)
```

Comment:
you could make this a bit more readable: `for (const auto & it_satellite : satellites) `

Reply:
@hroest Thanks for pointing this out. Very neat.